### PR TITLE
Explicitly set OSX_DEPLOYMENT_TARGET in cmake flags, as matrix.config.env is not propagated into job's env variables.

### PR DESCRIPTION
**Pull request by kadircet**
Fixes https://github.com/clangd/vscode-clangd/issues/228

### CODE REVIEW

**Comment by kadircet** [comment]
According to objdump min macosx version is now 10.9:
```
Load command 9
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.9
      sdk 11.1
```

File: .github/workflows/autobuild.yaml [hunk 1]
```diff
@@ -160,15 +160,16 @@ jobs:
...
           cmake: >-
             "-DCMAKE_C_COMPILER=clang"
             "-DCMAKE_CXX_COMPILER=clang++"
             "-DLLVM_ENABLE_ZLIB=FORCE_ON"
             "-DCMAKE_OSX_ARCHITECTURES=x86_64;arm64"
+            "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.9"
           # BoringSSL doesn't support universal binaries when building with ASM.
           grpc_cmake: >-
             "-DOPENSSL_NO_ASM=ON"
+            "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.9"
```
Comment (sam-mccall) [inline .github/workflows/autobuild.yaml:172]: I don't think you need it here, $grpc_cmake is *additions* to $cmake.

**Review by sam-mccall — APPROVED** [review]
Thanks! We could also propagate matrix.env into the environment, but this seems less magic.

File: .github/workflows/autobuild.yaml (hunk 1 above)
Comment (kadircet) [reply .github/workflows/autobuild.yaml:172]: right, thanks!

**Comment by kadircet** [comment]
> Thanks! We could also propagate matrix.env into the environment, but this seems less magic.

yeah i tried that too but couldn't find a nice way to represent that in the yaml file, especially when the matrix.env could be empty. also not relying on env variables sounded more robust in the end.

**Pull request merged**